addons/io/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/NeeoItemValueConverter.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.io.neeo.internal;
+
+import java.util.Objects;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.items.Item;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.library.types.UpDownType;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.Type;
+import org.eclipse.smarthome.core.types.TypeParser;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.io.neeo.NeeoUtil;
+import org.openhab.io.neeo.internal.servletservices.DeviceService;
+import org.openhab.io.neeo.internal.servletservices.models.PathInfo;
+import org.openhab.io.neeo.models.NeeoDeviceChannel;
+import org.openhab.io.neeo.models.NeeoItemValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The class is responsible for formatting a {@link State} to a string value. A lot of the code was 'borrowed' from
+ * ItemUIRegistryImpl.java
+ *
+ * @author Tim Roberts - Initial contribution
+ */
+public class NeeoItemValueConverter {
+
+    /** RegEx to identify format patterns. See java.util.Formatter#formatSpecifier (without the '%' at the very end). */
+    protected static final String IDENTIFY_FORMAT_PATTERN_PATTERN = "%(\\d+\\$)?([-#+ 0,(\\<]*)?(\\d+)?(\\.\\d+)?([tT])?([a-zA-Z])";
+
+    /**
+     * Converts the {@link State} from the given {@link NeeoDeviceChannel} to a {@link NeeoItemValue}. This method will
+     * convert enums to booleans and format string values to their associated pattern
+     *
+     * @param channel the non-null channel
+     * @param state the non-null state
+     * @return the non-null {@link NeeoItemValue}
+     */
+    public static NeeoItemValue convert(NeeoDeviceChannel channel, State state) {
+        Objects.requireNonNull(channel, "channel cannot be null");
+        Objects.requireNonNull(state, "state cannot be null");
+
+        final String format = channel.getValue();
+        if (state instanceof OnOffType) {
+            return new NeeoItemValue(state == OnOffType.ON ? true : false);
```

Comment:
This can be simplified to `return new NeeoItemValue(state == OnOffType.ON);`  Same for the other statements below.

Reply:
Wow - idiot move on my part - don't even know what I was thinking!  ;)